dot/network/service.go

```diff
@@ -711,7 +708,7 @@ func (s *Service) processMessage(msg peerset.Message) {
 		addrInfo := s.host.p2pHost.Peerstore().PeerInfo(peerID)
 		if len(addrInfo.Addrs) == 0 {
 			var err error
-			addrInfo, err = s.host.discovery.findPeer(peerID)
+			addrInfo, err = s.host.discovery.dht.FindPeer(s.host.discovery.ctx, peerID)
```

Comment:
I would suggest to create a new context with timeout here then we have a limit while finding a peer
```suggestion
                         ctx, cancel := context.WithTimeout(s.host.discovery.ctx, _some_timeout_limit_)
                         defer cancel()
			addrInfo, err = s.host.discovery.dht.FindPeer(ctx, peerID)
```